### PR TITLE
Add powered by Mapbox

### PR DESCRIPTION
**Pull request by mapbox-danny**
- [x] Add "powered by Mapbox" notation to the results list 
 - [ ] write tests for all new functionality
 - [x] run `npm run docs` and commit changes to API.md
 - [x] update CHANGELOG.md with changes under `master` heading before merging

### CODE REVIEW

**Comment by mbullington** [comment]
Screen Shot 2022-01-25 at 3 47 40 PM

To make the text nicer I also ended up upgrading the system font stack, so it includes Apple's San Francisco font if available. This was surely the intent originally, but San Francisco was first released near when this started development.

We also use a similar font stack in GL JS now: https://github.com/mapbox/mapbox-gl-js/pull/11029

**Review by mpothier — APPROVED** [review]

File: lib/index.js [hunk 1]
```diff
@@ -18,6 +18,17 @@ const GEOCODE_REQUEST_TYPE = {
   REVERSE: 2,
 };
 
+/**
+ * Don't include this as part of the options object when creating a new MapboxGeocoder instance.
+ */
+function getFooterNode() {
+  var div = document.createElement('div');
+  div.className = 'mapboxgl-ctrl-geocoder--powered-by';
+  div.innerHTML = '<a href="https://mapbox.com">Powered by Mapbox</a>';
```
[the remark this answers is not in the document]
Comment (mbullington) [reply lib/index.js:27]: Good idea

File: .vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+  "editor.formatOnSave": false,
+  "editor.codeActionsOnSave": {
+    "source.fixAll.eslint": false
+  }
+}
```
[the remark this answers is not in the document]
Comment (mbullington) [reply .vscode/settings.json:6]: Yes, I can omit it though if that makes more sense. Our ESLint version is so old I wasn't able to get it to work with the VSCode extension, and it was loading the config from another repo in my project folder.

[the remark this answers is not in the document]
Comment (mpothier) [reply .vscode/settings.json:6]: Ok yeah I figured it was something like that, I'm good with it

**Pull request merged**